modules/engines/engine-core/src/test/java/com/sap/xsk/modificators/XSKProjectFilesModificatorTest.java

```diff
@@ -0,0 +1,224 @@
+        StandardCharsets.UTF_8).getBytes();
+    IFile hdbtablefunctionIFile = project.createFile("test.hdbtablefunction", hdbtablefunctionFile);
+
+    byte[] analyticPrivilegeFile = IOUtils.toString(
+        XSKProjectFilesModificatorTest.class.getResourceAsStream("/META-INF/modificators/files-to-modify/test.analyticprivilege"),
+        StandardCharsets.UTF_8).getBytes();
+    IFile analyticPrivilegeIFile = project.createFile("test.analyticprivilege", analyticPrivilegeFile);
+
+    byte[] hdbanalyticPrivilegeFile = IOUtils.toString(
+        XSKProjectFilesModificatorTest.class.getResourceAsStream("/META-INF/modificators/files-to-modify/test.hdbanalyticprivilege"),
+        StandardCharsets.UTF_8).getBytes();
+    IFile hdbanalyticPrivilegeIFile = project.createFile("test.hdbanalyticprivilege", hdbanalyticPrivilegeFile);
+
+    projectFiles.add(xsjsIFile);
+    projectFiles.add(xsjslibIFile);
+    projectFiles.add(hdbprocedureIFile);
+    projectFiles.add(hdbtablefunctionIFile);
+    projectFiles.add(analyticPrivilegeIFile);
+    projectFiles.add(hdbanalyticPrivilegeIFile);
+
+    projectFilesModificator.modifyProjectFiles(project, projectFiles);
+
+    byte[] expectedModifiedXsjsFile = XSKProjectFilesModificatorTest.class.getResourceAsStream(
+        "/META-INF/modificators/expected-results/test.xsjs").readAllBytes();
+    byte[] expectedModifiedXsjslibFile = XSKProjectFilesModificatorTest.class.getResourceAsStream(
+        "/META-INF/modificators/expected-results/test.xsjslib").readAllBytes();
+    byte[] expectedModifiedHdbProcedureFile = XSKProjectFilesModificatorTest.class.getResourceAsStream(
+        "/META-INF/modificators/expected-results/test.hdbprocedure").readAllBytes();
+    byte[] expectedModifiedHdbtablefunctionFile = XSKProjectFilesModificatorTest.class.getResourceAsStream(
+        "/META-INF/modificators/expected-results/test.hdbtablefunction").readAllBytes();
+    byte[] expectedModifiedAnalyticprivilegeFile = XSKProjectFilesModificatorTest.class.getResourceAsStream(
+        "/META-INF/modificators/expected-results/test.analyticprivilege").readAllBytes();
+    byte[] expectedModifiedHdbanalyticprivilegeFile = XSKProjectFilesModificatorTest.class.getResourceAsStream(
+        "/META-INF/modificators/expected-results/test.hdbanalyticprivilege").readAllBytes();
+
+    for (IFile projectFile : projectFiles) {
+      switch (projectFile.getName()) {
+        case "test.xsjs": {
+          assertArrayEquals(projectFile.getContent(), expectedModifiedXsjsFile);
+          break;
+        }
+        case "test.xsjslib": {
+          assertArrayEquals(projectFile.getContent(), expectedModifiedXsjslibFile);
+          break;
+        }
+        case "test.hdbprocedure": {
+          assertArrayEquals(projectFile.getContent(), expectedModifiedHdbProcedureFile);
+          break;
+        }
+        case "test.hdbtablefunction": {
+          assertArrayEquals(projectFile.getContent(), expectedModifiedHdbtablefunctionFile);
```

Comment:
it is a good practice to add message to the asserts because you can have several in one test methods and when it fails you will have message that will tell you what is not as expected and not just some json/xml/free text that is not matching with some other json/xml/free text and this will help to support/maintain and faster troubleshot.